src/Jackalope/Node.php

```diff
@@ -773,9 +785,10 @@ public function getPropertiesValues($nameFilter = null, $dereference = true)
         $this->checkState();
 
         // OPTIMIZE: do not create properties in constructor, go over array here
-        $names = self::filterNames($nameFilter, array_keys($this->properties));
+        $names = self::filterNames($nameFilter, array_keys($this->propertyValues));
         $result = array();
         foreach ($names as $name) {
+            $this->loadProperty($name);
```

Comment:
Since afterwards `$this->properties[$name]` is accessed, I was assuming that the property was already instantiated. And since before this PR all properties have been instantiated eagerly, this could have not been the purpose of this method (or at least it didn't work that way), or am I missing something?